src/inet/InetInterface.h

```diff
@@ -540,15 +540,13 @@ class DLL_EXPORT InterfaceAddressIterator
     int mCurAddrIndex   = -1;
 #endif // CHIP_SYSTEM_CONFIG_USE_ZEPHYR_NET_IF
 #if CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT
-    otIp6AddressInfo * mAddrInfoList;
-    int mCurAddrIndex;
-    InterfaceIterator mIntfIter;
+    const otNetifAddress * mNetifAddrList;
+    otNetifAddress * mCurAddr;
```

Comment:
```suggestion
    const otNetifAddress * mCurAddr;
```
and then I think the const_cast can go away.

Reply:
fixed